File: src/main/java/nl/fizzylogic/intentify/entities/ValidationError.java
package nl.fizzylogic.intentify.entities;

import java.util.List;

/**
 * Represents a set of validation constraint violations
 */
public class ValidationError {
    private final List<FieldError> errors;

    /**
     * Initializes a new instance of {@link ValidationError}
     * @param errors
     */
    public ValidationError(List<FieldError> errors) {
        this.errors = errors;
    }

    /**
     * Gets the errors found in the input
     * @return  The list of errors for each field
     */
    public List<FieldError> getErrors() {
        return errors;
    }
}

File: src/main/kotlin/nl/fizzylogic/intentify/entities/ValidationError.kt
package nl.fizzylogic.intentify.entities

import org.springframework.validation.FieldError

/**
 * Represents a validation error
 */
class ValidationError {
    /**
     * Initializes a new instance of ValidationError
     * @param allErrors The errors that were detected by the controller
     */
    constructor(allErrors: List<FieldError>) {
        errors = allErrors.map { element ->
            FieldErrorMessage(element.field, element.defaultMessage)
        }.toList()
    }

    /**
     * Gets the errors that occurred
     */
    val errors: List<FieldErrorMessage>
}